### ISSUE TITLE
Segmentation fault in ExchangeClient::backgroundCpuTimeMs() 

### ISSUE THREAD

**Issue by karteekmurthys** [facebookincubator/velox#6504]
### Bug description

While testing with large scale sf10000 workload, we hit this issue in Prestissimo:

```
2023-09-09T01:03:13.355-07:00 | *** SIGSEGV (@0x31325f6d75bb) received by PID 16 (TID 0x7db3b8dfa700) from PID 1601009083; stack trace: ***
-- | --
  | 2023-09-09T01:03:13.357-07:00 | @ 0x7fb9a741e48e google::(anonymous namespace)::FailureSignalHandler()
  | 2023-09-09T01:03:13.359-07:00 | @ 0x7fb9a3ac1cf0 (unknown)
  | 2023-09-09T01:03:13.364-07:00 | @ 0x3c79c5f facebook::velox::exec::ExchangeClient::backgroundCpuTimeMs()
  | 2023-09-09T01:03:13.375-07:00 | @ 0x254bf65 facebook::velox::exec::Operator::close()
  | 2023-09-09T01:03:13.383-07:00 | @ 0x3c7697e facebook::velox::exec::Exchange::close()
  | 2023-09-09T01:03:13.390-07:00 | @ 0x3c6c435 facebook::velox::exec::Driver::closeOperators()
  | 2023-09-09T01:03:13.396-07:00 | @ 0x3c6d0c1 facebook::velox::exec::Driver::close()
  | 2023-09-09T01:03:13.404-07:00 | @ 0x3c71998 facebook::velox::exec::Driver::runInternal()
  | 2023-09-09T01:03:13.408-07:00 | @ 0x3c72857 facebook::velox::exec::Driver::run()
  | 2023-09-09T01:03:13.411-07:00 | @ 0x3c729ff _ZN5folly6detail8function14FunctionTraitsIFvvEE9callSmallIZN8facebook5velox4exec6Driver7enqueueESt10shared_ptrIS9_EEUlvE_EEvRNS1_4DataE
  | 2023-09-09T01:03:13.418-07:00 | @ 0x7fb9a5850bd9 folly::detail::function::FunctionTraits<>::operator()()
  | 2023-09-09T01:03:13.429-07:00 | @ 0x7fb9a0dd4954 folly::ThreadPoolExecutor::runTask()
  | 2023-09-09T01:03:13.432-07:00 | @ 0x7fb9a0d55330 folly::CPUThreadPoolExecutor::threadRun()
  | 2023-09-09T01:03:13.436-07:00 | @ 0x7fb9a0ddd055 std::__invoke_impl<>()
  | 2023-09-09T01:03:13.438-07:00 | @ 0x7fb9a0ddc583 std::__invoke<>()
  | 2023-09-09T01:03:13.441-07:00 | @ 0x7fb9a0ddb8b3 _ZNSt5_BindIFMN5folly18ThreadPoolExecutorEFvSt10shared_ptrINS1_6ThreadEEEPS1_S4_EE6__callIvJEJLm0ELm1EEEET_OSt5tupleIJDpT0_EESt12_Index_tupleIJXspT1_EEE
  | 2023-09-09T01:03:13.446-07:00 | @ 0x7fb9a0dda956 std::_Bind<>::operator()<>()
  | 2023-09-09T0
```

There were recent changes made to this section of code: https://github.com/facebookincubator/velox/pull/6352/

May be the `source` is cleaned up at time of accessing its `source->backgroundCpuTimeMs()`.
```
uint64_t ExchangeClient::backgroundCpuTimeMs() const {
  uint64_t backgroundCpuTimeMs = 0;
  for (const auto& source : sources_) {
    backgroundCpuTimeMs += source->backgroundCpuTimeMs();
  }

  return backgroundCpuTimeMs;
}
```

### System information

NA

**Comment by karteekmurthys** [comment]
cc: @majetideepak @aditi-pandit

**Comment by karteekmurthys** [comment]
cc: @patrickstuedi @mbasmanova

**Comment by patrickstuedi** [comment]
@karteekmurthys the PR (6352) causing this was reverted by this PR 6491

https://github.com/facebookincubator/velox/pull/6491

**Comment by mbasmanova** [comment]
Fixed in #6491

**Comment by patrickstuedi** [comment]
Advance velox version to include the fix #20832
